src/pages/index.js

```diff
@@ -184,11 +252,50 @@ cardModalCloseBtn.addEventListener("click", () => {
   closeModal(cardModal);
 });
 
+//functions for edit avatar modal//
+editAvatarText.addEventListener("click", () => {
+  resetValidation(editAvatarForm, [editAvatarInput], settings);
+  openModal(editAvatarModal);
+});
+
+editAvatarContainer.addEventListener("click", () => {
+  resetValidation(editAvatarForm, [editAvatarInput], settings);
+  openModal(editAvatarModal);
+});
+
+editAvatarCloseBtn.addEventListener("click", () => {
+  closeModal(editAvatarModal);
+});
+
+function handleEditAvatarSubmit(evt) {
+  evt.preventDefault();
+  const originalTextEditAvatar = editAvatarBtn.textContent;
+  editAvatarBtn.textContent = "Saving...";
+  api
+    .editAvatar({ avatar: editAvatarInput.value })
+    .then((data) => {
+      profileAvatar.src = data.avatar;
+    })
+    .catch(console.error)
+    .finally(() => {
+      editAvatarBtn.textContent = originalTextEditAvatar;
+      closeModal(editAvatarModal);
```

Comment:
* You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**